src/casing.test.ts

```diff
@@ -156,4 +156,65 @@ describe('casing functions', () => {
     expect(result).toEqual(expected)
     type test = Expect<Equal<typeof result, typeof expected>>
   })
+
+  describe('with various separators', () => {
+    const text =
+      '[one] two-three/four.five(six){seven}|eight_nine\\ten' as const
+
```

Comment:
I wish for a DRYer way to write this test suite -- they're all the exact same.

There isn't a way to do the following, is there?

```ts
const text = 'abc'

const resultExpectedMap = new Map([
  [subject.toUpperCase(text), 'ABC'],
  // ...
]) 

for (const [result, expected] in  resultExpectedMap.entries() {
  expect(result).toEqual(expected)
  type test = Expect<Equal<typeof result, typeof expected>>
}
```

Reply:
The closest I could get is this
<img width="416" alt="Screenshot 2023-10-05 at 1 26 26 PM" src="https://github.com/gustavoguichard/string-ts/assets/33426811/acd9eef5-0537-4342-8eae-3e95bf94f80a">

Reply:
The problem is that I could switch `ABC` and `abc` on the expected side and the type union would still be equal

Reply:
It could be easier... I personally am not a fan of too much abstractions in tests.. I tend to just repeat LOCs to avoid potential footguns.
But not against the trials